### ISSUE TITLE
NPE in HandleSuperBuilder

### ISSUE THREAD

**Issue by cameronbraid** [projectlombok/lombok#2534]
```
[Error - 12:14:52 AM] 4 Aug. 2020, 12:14:52 am Lombok annotation handler class lombok.eclipse.handlers.HandleSuperBuilder failed
null
java.lang.NullPointerException
	at lombok.eclipse.handlers.HandleSuperBuilder.gatherUsedTypeNames(HandleSuperBuilder.java:1069)
	at lombok.eclipse.handlers.HandleSuperBuilder.handle(HandleSuperBuilder.java:229)
	at lombok.eclipse.HandlerLibrary$AnnotationHandlerContainer.handle(HandlerLibrary.java:106)
	at lombok.eclipse.HandlerLibrary.handleAnnotation(HandlerLibrary.java:237)
	at lombok.eclipse.TransformEclipseAST$AnnotationVisitor.visitAnnotationOnType(TransformEclipseAST.java:238)
	at lombok.eclipse.EclipseNode.traverse(EclipseNode.java:106)
	at lombok.eclipse.EclipseAST.traverseChildren(EclipseAST.java:233)
	at lombok.eclipse.EclipseNode.traverse(EclipseNode.java:73)
	at lombok.eclipse.EclipseAST.traverseChildren(EclipseAST.java:233)
	at lombok.eclipse.EclipseNode.traverse(EclipseNode.java:68)
	at lombok.eclipse.EclipseAST.traverse(EclipseAST.java:226)
	at lombok.eclipse.TransformEclipseAST.go(TransformEclipseAST.java:195)
	at lombok.eclipse.TransformEclipseAST.transform(TransformEclipseAST.java:156)
	at lombok.eclipse.TransformEclipseAST.transform_swapped(TransformEclipseAST.java:106)
	at jdk.internal.reflect.GeneratedMethodAccessor8.invoke(Unknown Source)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at lombok.launch.PatchFixesHider$Util.invokeMethod(PatchFixesHider.java:115)
	at lombok.launch.PatchFixesHider$Transform.transform_swapped(PatchFixesHider.java:186)
	at org.eclipse.jdt.internal.compiler.parser.Parser.endParse(Parser.java:11623)
	at org.eclipse.jdt.internal.core.util.CommentRecorderParser.endParse(CommentRecorderParser.java:135)
	at org.eclipse.jdt.internal.compiler.parser.Parser.parse(Parser.java:12809)
	at org.eclipse.jdt.internal.compiler.parser.Parser.parse(Parser.java:13033)
	at org.eclipse.jdt.internal.compiler.parser.Parser.parse(Parser.java:12990)
	at org.eclipse.jdt.internal.compiler.parser.Parser.dietParse(Parser.java:11384)
	at org.eclipse.jdt.internal.compiler.Compiler.internalBeginToCompile(Compiler.java:850)
	at org.eclipse.jdt.internal.compiler.Compiler.beginToCompile(Compiler.java:394)
	at org.eclipse.jdt.internal.compiler.Compiler.resolve(Compiler.java:1042)
	at org.eclipse.jdt.internal.compiler.Compiler.resolve(Compiler.java:1115)
	at org.eclipse.jdt.internal.core.CompilationUnitProblemFinder.process(CompilationUnitProblemFinder.java:280)
	at org.eclipse.jdt.internal.core.CompilationUnitProblemFinder.process(CompilationUnitProblemFinder.java:346)
	at org.eclipse.jdt.internal.core.ReconcileWorkingCopyOperation.makeConsistent(ReconcileWorkingCopyOperation.java:193)
	at org.eclipse.jdt.internal.core.ReconcileWorkingCopyOperation.executeOperation(ReconcileWorkingCopyOperation.java:103)
	at org.eclipse.jdt.internal.core.JavaModelOperation.run(JavaModelOperation.java:736)
	at org.eclipse.jdt.internal.core.JavaModelOperation.runOperation(JavaModelOperation.java:802)
	at org.eclipse.jdt.internal.core.CompilationUnit.reconcile(CompilationUnit.java:1315)
	at org.eclipse.jdt.ls.core.internal.handlers.BaseDocumentLifeCycleHandler.publishDiagnostics(BaseDocumentLifeCycleHandler.java:264)
	at org.eclipse.jdt.ls.core.internal.handlers.BaseDocumentLifeCycleHandler.publishDiagnostics(BaseDocumentLifeCycleHandler.java:228)
	at org.eclipse.jdt.ls.core.internal.handlers.BaseDocumentLifeCycleHandler$2.runInWorkspace(BaseDocumentLifeCycleHandler.java:100)
	at org.eclipse.core.internal.resources.InternalWorkspaceJob.run(InternalWorkspaceJob.java:42)
	at org.eclipse.core.internal.jobs.Worker.run(Worker.java:63)
```

**Version info (please complete the following information):**
 - Lombok version
1.18.12

 - Platform (javac or eclipse, and if so, what is the output of `javac -version` / the version listed in the _about..._ dialog of eclipse.
vscode
openjdk version "11.0.8" 2020-07-14
OpenJDK Runtime Environment (build 11.0.8+10-post-Ubuntu-0ubuntu120.04)
OpenJDK 64-Bit Server VM (build 11.0.8+10-post-Ubuntu-0ubuntu120.04, mixed mode, sharing)

**Comment by cameronbraid** [comment]
using vscode lombok v 1.0.1

**Comment by janrieke** [comment]
Could you add the code where this issue occurs?

**Comment by cameronbraid** [comment]
The lombok version I stated initiall is in my pom.xml.. but looking at the source of HandleSuperBuilder, there is no line 1069 so I am a bit confused

**Comment by cameronbraid** [comment]
can I add the code - no sorry, but I can sketch an outline - the issue is with class Supplier below

```
@TypeDefs({
    @TypeDef(name = "json", typeClass = JsonStringType.class)
})
@MappedSuperclass
@SuperBuilder(toBuilder = true)
@AllArgsConstructor
@NoArgsConstructor
public abstract class AbstractIdEntity implements Serializable {
...
}
```

```

@Entity
@DynamicUpdate
@DynamicInsert
@Table(name = "ut_supplier")
@Cache(usage = CacheConcurrencyStrategy.NONSTRICT_READ_WRITE)
@RecordModelRef(name = "supplier")
@SuperBuilder(toBuilder = true)
@NoArgsConstructor
@AllArgsConstructor
public class Supplier extends AbstractIdEntity implements HasInternalName, HasInternalCode {
...
}
```

I have other subclasses of AbstractIdEntity and their @SuperBuilder is woking fine.  Im working on trying to identify the specific differences.

**Comment by cameronbraid** [comment]
I think I found it

at the start of Supplier class it has 

```
  static {
    EntityPermission.registerEntityType("supplier", Supplier.class, "internalCode");
  }
```

When I comment it out the builder is generated

**Comment by Rawi01** [comment]
Duplicate of #2716 and already fixed in 1.18.20
